tests/visualization_tests/test_contour.py

```diff
@@ -67,219 +84,279 @@ def fail_objective(_: Trial) -> float:
 
     study = create_study()
     study.optimize(fail_objective, n_trials=1, catch=(ValueError,))
-    figure = plot_contour(study, params=params)
-    assert len(figure.data) == 0
+    figure = create_figure(study, params=params)
+    assert figure.is_empty()
+    figure.save_static_image()
 
     # Test with some trials.
     study = prepare_study_with_trials()
 
     # Test ValueError due to wrong params.
     with pytest.raises(ValueError):
-        plot_contour(study, ["optuna", "Optuna"])
+        _ = create_figure(study, ["optuna", "Optuna"])
 
-    figure = plot_contour(study, params=params)
+    figure = create_figure(study, params=params)
     if params is not None and len(params) < 3:
         if len(params) <= 1:
-            assert not figure.data
+            assert figure.is_empty()
         elif len(params) == 2:
-            assert figure.data[0]["x"] == (0.925, 1.0, 2.5, 2.575)
-            assert figure.data[0]["y"] == (-0.1, 0.0, 1.0, 2.0, 2.1)
-            assert figure.data[0]["z"][3][1] == 0.0
-            assert figure.data[0]["z"][2][2] == 1.0
-            assert figure.data[0]["colorbar"]["title"]["text"] == "Objective Value"
-            assert figure.data[1]["x"] == (1.0, 2.5)
-            assert figure.data[1]["y"] == (2.0, 1.0)
-            assert figure.layout["xaxis"]["range"] == (0.925, 2.575)
-            assert figure.layout["yaxis"]["range"] == (-0.1, 2.1)
+            if isinstance(figure, PlotlyContourFigure):
```

Comment:
Because matplotlib has no interface to get the grid and map, I use an ugly branching here and others. We have some choices to avoid this.
(1) Make the `plot_contour` itself testable. This is an alternative in the PR description.
(2) Discard tests where the API to retrieve only exists on one side.
What do you think?

Reply:
Hmm, it sounds tough to decide the approach, but I suppose the current approach looks great. 

Option (2) sounds reasonable somehow but it would be more robust to have at least either backend's test.